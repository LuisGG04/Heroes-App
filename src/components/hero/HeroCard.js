import React from "react";
import { Link } from "react-router-dom";

const heroImages = require.context("../../assets", true);

export const HeroCard = ({id,
    superhero,
    publisher,
    alter_ego,
    first_appearance,
    characters}) => {

    // const imagePath = `/assets/${id}.jpg`

    return (
        <article className="col animate__animated animate__fadeIn">
            <div className="card">
                <div className="row no-gutters">
                    <div className="col-4">
                        <img src={heroImages(`./${id}.jpg`)} className="card-img" alt={superhero}/>
                    </div>
                    <div className="col-8">
                        <div className="card-body">
                            <h3 className="card-title">{superhero}</h3>
                            <p className="card-text">{alter_ego}</p>

                            {(alter_ego !== characters) && 
                            <p className="text-muted">{characters}</p>}

                            <p className="card-text"><small className="text-muted">{first_appearance}</small></p>

                            <Link to={`/hero/${id}`}>Más información</Link> 
                        </div>
                    </div>
                </div>
            </div>
        </article>
    )
}
