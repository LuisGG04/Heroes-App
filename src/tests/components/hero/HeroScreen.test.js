import { mount } from "enzyme";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import { HeroScreen } from "../../../components/hero/HeroScreen";

const mockNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useNavigate: () => mockNavigate
}));

describe("Pruebas en el componente HeroScreen", () => {

    test("No debe de mostrar el HeroScreen si no se tiene algún héroe como parámetro", () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={["/hero"]}>
                <Routes>
                    <Route path="/hero" element={<HeroScreen />} />
                    <Route path="/" element={<h1>No hero page</h1>} />
                </Routes>
            </MemoryRouter>
        );

        expect(wrapper.find("h1").text().trim()).toBe("No hero page");
    });

    test("Debe de mostrar un héroe si el parámetro existe y se encuentra", () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={["/hero/marvel-spider"]}>
                <Routes>
                    <Route path="/hero/:heroeId" element={<HeroScreen />} />
                    <Route path="/" element={<h1>No hero page</h1>} />
                </Routes>
            </MemoryRouter>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find(".row").exists()).toBe(true);
    });

    test("Debe de regresar a la pantalla anterior", () => {
        const wrapper = mount(
            <MemoryRouter initialEntries={["/hero/marvel-spider"]}>
                <Routes>
                    <Route path="/hero/:heroeId" element={<HeroScreen />} />
                </Routes>
            </MemoryRouter>
        );

        wrapper.find("button").prop("onClick")();

        expect(mockNavigate).toHaveBeenCalledWith(-1);
    });
    
});
