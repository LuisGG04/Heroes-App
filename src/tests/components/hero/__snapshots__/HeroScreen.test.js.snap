// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente HeroScreen Debe de mostrar un héroe si el parámetro existe y se encuentra 1`] = `
<article
  className="row mt-5"
>
  <div
    className="col-4"
  >
    <img
      alt="Spider Man"
      className="img-thumbnail animate__animated animate__fadeInLeft"
      src="/assets/marvel-spider.jpg"
    />
  </div>
  <div
    className="col-8 animate__animated animate__fadeIn"
  >
    <h3>
      Spider Man
    </h3>
    <ul
      className="list-group"
    >
      <li
        className="list-group-item"
      >
        <b>
          Alter ego:
        </b>
         
        Peter Parker
      </li>
      <li
        className="list-group-item"
      >
        <b>
          Publisher:
        </b>
         
        Marvel Comics
      </li>
      <li
        className="list-group-item"
      >
        <b>
          First appearance:
        </b>
         
        Amazing Fantasy #15
      </li>
    </ul>
    <h4
      className="mt-4"
    >
      Characters
    </h4>
    <p>
      Peter Parker
    </p>
    <button
      className="btn btn-outline-info"
      onClick={[Function]}
    >
      Regresar
    </button>
  </div>
</article>
`;
